lib/wamp_proto/message/challenge.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module WampProto
+  module Message
+    # Wamp Challenge message
+    class Challenge < Base
+      attr_reader :auth_method, :extra
+
+      def initialize(auth_method, extra = {})
+        super()
+        @auth_method = Validate.string!("AuthMethod", auth_method)
+        @extra = Validate.hash!("Extra", extra)
+      end
+
+      def payload
```

Comment:
```suggestion
      def marshal
```

This should be marshal everywhere

Reply:
@om26er  updated the code